src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java

```diff
@@ -255,4 +266,99 @@ private static Stream<Arguments> invalidNames() {
         Arguments.of("Te*st")
     );
   }
+
+  @Test
+  public void requestRoleChange_valid() throws NoCredentialsAvailableException, NoUserException, SegueDatabaseException,
+      AuthenticationProviderMappingException, IncorrectCredentialsProvidedException,
+      AdditionalAuthenticationRequiredException, InvalidKeySpecException, NoSuchAlgorithmException,
+      MFARequiredButNotConfiguredException {
+    LoginResult currentlyStudentLogin = loginAs(httpSession, ITConstants.TEST_STUDENT_EMAIL, ITConstants.TEST_STUDENT_PASSWORD);
+    HttpServletRequest upgradeRequest = createRequestWithCookies(new Cookie[] {currentlyStudentLogin.cookie});
+    replay(upgradeRequest);
+    Map<String, String> requestDetails = Map.of(
+        "verificationDetails", "school staff url",
+        "otherDetails", "more information"
+    );
+    RegisteredUserDTO initialUserState = currentlyStudentLogin.user;
+    assertFalse(initialUserState.getTeacherPending());
+    initialUserState.setTeacherPending(true);
```

Comment:
I think you've done this in order to later on compare this initial user state with the updated response, to ensure that the teacher pending flag is `true` after the role change request. But this is very confusing, it's very hard to understand your intention, so this needs to be rewritten. You shouldn't be setting this manually. All you need to do is to assert that the flag is true, but using the entity from the upgraded response.